src/Grpc.Net.Client/Internal/Retry/HedgingCall.cs

```diff
@@ -60,151 +60,157 @@ public HedgingCall(HedgingPolicyInfo hedgingPolicy, GrpcChannel channel, Method<
 
         private async Task StartCall(Action<GrpcCall<TRequest, TResponse>> startCallFunc)
         {
-            GrpcCall<TRequest, TResponse> call;
-            lock (Lock)
+            GrpcCall<TRequest, TResponse>? call = null;
+
+            try
             {
-                if (CommitedCallTask.IsCompletedSuccessfully())
+                lock (Lock)
                 {
-                    // Call has already been commited. This could happen if written messages exceed
-                    // buffer limits, which causes the call to immediately become commited and to clear buffers.
-                    return;
-                }
+                    if (CommitedCallTask.IsCompletedSuccessfully())
+                    {
+                        // Call has already been commited. This could happen if written messages exceed
+                        // buffer limits, which causes the call to immediately become commited and to clear buffers.
+                        return;
+                    }
 
-                OnStartingAttempt();
+                    OnStartingAttempt();
 
-                call = HttpClientCallInvoker.CreateGrpcCall<TRequest, TResponse>(Channel, Method, Options, AttemptCount);
-                _activeCalls.Add(call);
+                    call = HttpClientCallInvoker.CreateGrpcCall<TRequest, TResponse>(Channel, Method, Options, AttemptCount);
+                    _activeCalls.Add(call);
 
-                startCallFunc(call);
+                    startCallFunc(call);
 
-                SetNewActiveCallUnsynchronized(call);
-            }
+                    SetNewActiveCallUnsynchronized(call);
+                }
 
-            Status? responseStatus;
+                Status? responseStatus;
 
-            HttpResponseMessage? httpResponse = null;
-            try
-            {
-                if (call._httpResponseTask == null)
+                HttpResponseMessage? httpResponse = null;
+                try
                 {
-                    // There is no response task if there was a preemptive cancel.
-                    CompatibilityHelpers.Assert(call.CancellationToken.IsCancellationRequested, "Request should have been made if call is not preemptively cancelled.");
-                    call.CancellationToken.ThrowIfCancellationRequested();
+                    httpResponse = await call.HttpResponseTask.ConfigureAwait(false);
```

Comment:
```suggestion
                    httpResponse = await call!.HttpResponseTask.ConfigureAwait(false);
```

`call` is guaranteed to be set here, correct? Annotation makes it a little easier to read since it's nullable.

Reply:
Yes, a call is always set at this point.

I think an annotation makes it less clear what is going on. Since this compiles, code analysis guarantes that it's not null here. An annotation would make me wonder why this could be null.